### PR TITLE
feat (migration to typescript): transform javascript to typescript

### PR DESCRIPTION
**Pull request by dependentmadani**
Migrate from Javascript to Typescript.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,32 +1,82 @@
...
+  /** Embedded content height */
+  height?: number;
+}
+
 /**
- * @typedef {object} EmbedConfig
  * @description Embed tool configuration object
- * @property {object} [services] - additional services provided by user. Each property should contain Service object
  */
+interface EmbedConfig {
```
Comment (neSpecc) [inline src/index.ts:45]: lets export it as well as `Service`

File: src/index.ts [hunk 2]
```diff
@@ -1,32 +1,82 @@
...
- * @typedef {object} PasteEvent
- * @typedef {object} HTMLElement
- * @typedef {object} Service
  * @description Service configuration object
- * @property {RegExp} regex - pattern of source URLs
- * @property {string} embedUrl - URL scheme to embedded page. Use '<%= remote_id %>' to define a place to insert resource id
- * @property {string} html - iframe which contains embedded content
- * @property {Function} [id] - function to get resource id from RegExp groups
  */
+interface Service {
```
Comment (neSpecc) [inline src/index.ts:27]:
```suggestion
interface ServiceConfig {
```

File: src/services.ts [hunk 3]
```diff
@@ -1,5 +1,14 @@
 /* eslint-disable no-useless-escape */
-export default {
+interface Service {
```
Comment (neSpecc) [inline src/services.ts:2]: it repeats the interface from index.ts. You can move types to a separate file and depend on it from both files

File: src/index.ts [hunk 4]
```diff
@@ -1,32 +1,82 @@
...
+}
+
 /**
- * @typedef {object} EmbedConfig
  * @description Embed tool configuration object
- * @property {object} [services] - additional services provided by user. Each property should contain Service object
  */
+interface EmbedConfig {
+  /** Additional services provided by user */
+  services?: { [key: string]: Service | boolean };
```
Comment (neSpecc) [inline src/index.ts:47]: maybe we should move `{ [key: string]: Service | boolean }` to some type since it is used in many places

File: package.json
Comment (neSpecc) [inline package.json:1]: please, update a patch version

**Comment by dependentmadani** [comment]
## Changes made:
- Export necessary interfaces.
- Create type file and define the necessary.
- Convert test js file into typescript.

**Review by neSpecc — APPROVED** [review]

**Pull request merged**